docs/advanced/tls_hardware_acceleration.md

```diff
@@ -0,0 +1,131 @@
+
+mbed TLS has a variety of options to make use of your alternative implementation. These make it possible to easily replace functionality at various abstraction levels and to different extents. In other words, you can replace the least amount of code to reach the highest possible acceleration with the smallest amount of effort.
+
+## Adding acceleration by replacing functions
+
+First, you should consider what kind of functionality your hardware provides. Does the processor have some modular arithmetic capabilities? Or does your board have a full cryptographic module, securely storing keys and providing the functionality of high level cryptographic primitives?
+
+### Process overview
+
+1. Identify the module or the functions you want to replace. For example, if you have a full cryptographic module, then you probably want to replace functions in the modules. If you only have a couple of special instructions or a co-processor that accelerates some part of the cryptographic function, then you may want to replace only the relevant functions in the module.
+
+1. If you want to replace functions in the ECP module, you need to implement the mandatory utility functions:
+    - these are functions that do not have a counterpart in the standard mbed TLS implementation and their only purpose is to facilitate the integration of the accelerated functions.
+    - `mbedtls_internal_ecp_grp_capable`: implement it to tell mbed TLS if the cryptographic hardware can handle the group. If the answer is no, then mbed TLS will fall back to the software implementation to continue the operation.
+    - `mbedtls_internal_ecp_init` and `mbedtls_internal_ecp_free` are optional. Use them to optimise if you are replacing a function in the ECP module.
+    - For more information about the utility functions read the subsection about the [ECP](#How-to-implement-ECP-module-functions) module.
+
+1. Implement the selected functions with the help of your hardware accelerator and copy the source code to the `features/mbedtls/targets/TARGET_XXXX` directory specific to your target. You may create a directory structure similar to the one you have for the HAL if you feel it appropriate. These functions have the same name as the ones they replace. We have [doxygen documentation for the original functions](https://tls.mbed.org/api/). The exception to the naming conventions is the ECP module, where an internal API is exposed to enable hardware acceleration. These functions too have a doxygen documentation and you can find them in the `ecp_internal.h` header file.
+
+1. Since mbed TLS is a static link library you also have to somehow notify the compiler/linker that the alternative implementations are present. To do this, you have to set the macros corresponding to the selected functions. You can read more on this in the [subsection about setting macros](#How-to-set-the-macros?) section.
```

Comment:
'is a static link library' - yes in mbed OS. Can you change this to:

'Since mbed TLS is implemented as a static link library in mbed OS, you also have to....'